presto-orc/src/main/java/com/facebook/presto/orc/writer/DictionaryColumnWriter.java

```diff
@@ -286,7 +290,12 @@ public void writeBlock(Block block)
         ColumnStatistics statistics = createColumnStatistics();
         DictionaryRowGroup rowGroup = new DictionaryRowGroup(rowGroupIndexes, rowGroupValueCount, statistics);
         rowGroups.add(rowGroup);
-        rowGroupRetainedSizeInBytes += rowGroup.getRetainedSizeInBytes();
+        if (true == enableMemoryAccountingForSapphire) {
```

Comment:
if (enableMemoryAccountingForSapphire) should suffice.

Please also add a comment why we are re-introducing this bug and our plan to fix it in the future.

Reply:
This isn't Sapphire specific. It's actually the opposite it is Sapphire that wants the correct memory accounting it is some other niche (from a PrestoDB perspective) that prefer undercounting to breaking queries.

I think referring to it as loose memory accounting and documenting what that means might make sense.

Reply:
We should call this flag as includeRowGroupSize and just document it that, includingRowGroupSize is right it currently breaks queries.

Reply:
Is it really useful to start collecting N different options for undercounting memory?

Reply:
Today it maps to only one setting in the OrcWriter, but this fix is specific to not include RowGroupSize. Let us say, later, if we want to add do not include dictionary size ( just as an example), it will be a different setting for the writer. 

the other way is to keep one setting enableMemoryBasedAccounting and it can map to different properties in the writer. But that at the writer layer will be hard to understand.